### PR TITLE
Chatbot with full JWT verification

### PR DESCRIPTION
**Pull request by imaginal**

### CODE REVIEW

File: declarations_site/chatbot/utils.py [hunk 1]
```diff
@@ -35,10 +43,81 @@ def simple_search(query, deepsearch=False):
...
+
+    response = requests.get(openidURL, timeout=10)
+    config = response.json()
+
+    if config['issuer'] != 'https://api.botframework.com':
+        return False
+
+    response = requests.get(config['jwks_uri'], timeout=10)
+    keys = response.json()
+    if 'keys' in keys and keys['keys']:
```
Comment (dchaplinsky) [inline declarations_site/chatbot/utils.py]: if keys.get("keys")

File: declarations_site/chatbot/utils.py [hunk 2]
```diff
@@ -35,10 +43,81 @@ def simple_search(query, deepsearch=False):
...
+        cache.set(openidURL, keys, 86400)
```
Comment (dchaplinsky) [inline declarations_site/chatbot/utils.py]: Should we move that TTL and timeout=10 to the settings?

File: declarations_site/chatbot/utils.py [hunk 3]
```diff
@@ -35,10 +43,81 @@ def simple_search(query, deepsearch=False):
...
+
+
+def verify_jwt(auth, data):
+    if not settings.BOTAPI_JWT_VERIFY:
+        return True
+
+    try:
+        method, token = auth.split(' ')
+        token_fingerprint = sha1(token.encode()).hexdigest()
+    except:
```
Comment (dchaplinsky) [inline declarations_site/chatbot/utils.py]: Except what?

File: declarations_site/chatbot/utils.py (hunk 2 above)
Comment (imaginal) [reply declarations_site/chatbot/utils.py]: I do not see space for fine tuning here
In MS library these values are hardcoded

**Pull request merged**